types/node/v6/index.d.ts

```diff
@@ -1957,8 +1957,54 @@ declare module "url" {
     }
 
     export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): Url;
+    export function format(URL: URL, options?: URLFormatOptions): string;
     export function format(urlObject: UrlObject | string): string;
     export function resolve(from: string, to: string): string;
+
+    export function domainToASCII(domain: string): string;
+    export function domainToUnicode(domain: string): string;
+
+    export interface URLFormatOptions {
+        auth?: boolean;
+        fragment?: boolean;
+        search?: boolean;
+        unicode?: boolean;
+    }
+
+    export class URLSearchParams implements Iterable<string[]> {
```

Comment:
@Flarna I copied the types from those for Node v7 (as it was the closest version which had the WHATWG URL types). As such, if there’s an existing bogus mismatch between v7 and v10, it will be reflected here.

The change here appears to have been made in #19683 which updated the v8.x types (then the most recent), but neglected to update the v7 types. If it’s true that the v7 types were incorrectly not updated then, I can update the v7 URL to match v8, as well as the v6. Does it make sense as part of this P.R. or an additional P.R. which only addresses the inconsistency?

Reply:
@amarshall #19683 was several month after EOL of NodeJS 7 if I have the dates right. Therefore I assume no one cared much about it. In general fixes/improvements are often done only on latest version even if they would be applicable to other versions. Every now and then someone backports/syncs - at least for the recent versions.
Feel free to adapt v7 typings but I wound not expect someone is really using them these days (similar as v5 and in a few month v9).